plugin-name/uninstall.php

```diff
@@ -2,29 +2,60 @@
 /**
  * Fired when the plugin is uninstalled.
  *
+ * We are not using an uninstall hook because WordPress perfoms bad when using it.
+ * Even if below issue is "fixed", it did not resolve the perfomance issue.
+ *
+ * @see https://core.trac.wordpress.org/ticket/31792
+ *
+ *
  * When populating this file, consider the following flow
  * of control:
  *
- * - This method should be static
- * - Check if the $_REQUEST content actually is the plugin name
- * - Run an admin referrer check to make sure it goes through authentication
- * - Verify the output of $_GET makes sense
- * - Repeat with other user roles. Best directly by using the links/query string parameters.
- * - Repeat things for multisite. Once for a single site in the network, once sitewide.
- *
- * This file may be updated more in future version of the Boilerplate; however, this is the
- * general skeleton and outline for how the file should work.
+ * - Check if the $_REQUEST['plugin'] content actually is plugin-name/plugin-name.php
+ * - Check if the $_REQUEST['action'] content actually is delete-plugin
+ * - Run a check_ajax_referer check to make sure it goes through authentication
+ * - Run a current_user_can check to make sure current user can delete a plugin
  *
- * For more information, see the following discussion:
- * https://github.com/tommcfarlin/WordPress-Plugin-Boilerplate/pull/123#issuecomment-28541913
+ * @todo Consider multisite. Once for a single site in the network, once sitewide.
  *
  * @link       http://example.com
  * @since      1.0.0
- *
  * @package    Plugin_Name
  */
 
-// If uninstall not called from WordPress, then exit.
-if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
-	exit;
+/**
+ * Perform Uninstall Actions.
+ *
+ * If uninstall not called from WordPress,
+ * If no uninstall action,
+ * If not this plugin,
+ * If no caps,
+ * then exit.
+ *
+ * @since 1.0.0
+ */
+function plugin_name_uninstall() {
+
+	if ( ! defined( 'WP_UNINSTALL_PLUGIN' )
+		|| empty( $_REQUEST )
+		|| ! isset( $_REQUEST['plugin'] )
+		|| ! isset( $_REQUEST['action'] )
+		|| 'plugin-name/plugin-name.php' != $_REQUEST['plugin']
+		|| 'delete-plugin' != $_REQUEST['action']
```

Comment:
`Found: !=. Use strict comparisons (=== or !==) `
I do not consider this best practice. But I might be massively wrong here. And PHPCS complains, so... I should probably change that?

Reply:
It's usually better to use strict comparisons (===, !==) rather than loose comparisons (==, !=) because they're more explicit and therefore less prone to silly edge case bugs.

It is also considered a best practice in Javascript (I work full-time as a frontend-engineer).

Would recommend that you read more about it here: https://core.trac.wordpress.org/ticket/48965